frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1235,6 +1235,23 @@ export const surveyLogic = kea<surveyLogicType>([
                 return null
             },
         ],
+        deviceTypesMatchTypeValidationError: [
+            (s) => [s.survey],
+            (survey: Survey): string | null => {
+                if (
+                    survey.conditions?.deviceTypesMatchType === SurveyMatchType.Regex &&
+                    survey.conditions.deviceTypes
+                ) {
+                    try {
+                        new RegExp(survey.conditions.deviceTypes?.join('|'))
```

Comment:
this is wrong, the regex should be a single string, you don't add multiple regexes, you just set one that may contain multiple items in it but its still a single item|string